python_ta/checkers/global_variables_checker.py

```diff
@@ -76,6 +76,8 @@ def _get_child_disallowed_global_var_nodes(node):
     global, non-constant variable. 
     """
     node_list = []
+    if isinstance(node.parent, astroid.ListComp) or isinstance(node.parent, astroid.Comprehension):
```

Comment:
About the nesting: something like `[x + 3 for x in ...]` would have the `x` in `x+3` as a child of a `BinOp` node, right? If I understand this code properly, in this case the `x` in `x + 3` would be reported.